src/golang/lib/airflow/schedule.go

```diff
@@ -0,0 +1,297 @@
+		for _, artifactId := range op.Outputs {
+			dbOutputArtifact, ok := dag.Artifacts[artifactId]
+			if !ok {
+				return nil, errors.Newf("cannot find artifact with ID %v", artifactId)
+			}
+
+			contentPath := artifactToContentPathPrefix[artifactId]
+			metadataPath := artifactToMetadataPathPrefix[artifactId]
+
+			outputArtifact, err := artifact.NewArtifact(
+				dbOutputArtifact,
+				contentPath,
+				metadataPath,
+				nil,
+				storageConfig,
+				nil,
+			)
+			if err != nil {
+				return nil, err
+			}
+
+			outputArtifacts = append(outputArtifacts, outputArtifact)
+			outputContentPathPrefixes = append(outputContentPathPrefixes, contentPath)
+			outputMetadataPathPrefixes = append(outputMetadataPathPrefixes, metadataPath)
+		}
+
+		airflowOperator, err := operator.NewOperator(
+			ctx,
+			op,
+			inputArtifacts,
+			inputContentPathPrefixes,
+			inputMetadataPathPrefixes,
+			outputArtifacts,
+			outputContentPathPrefixes,
+			outputMetadataPathPrefixes,
+			nil,
+			jobManager,
+			vault,
+			&airflowStorageConfig,
+			db,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		jobSpec := airflowOperator.JobSpec()
+
+		taskId := generateTaskId(airflowOperator.Name())
+
+		operatorToTask[airflowOperator.ID()] = taskId
+		taskToJobSpec[taskId] = jobSpec
+	}
+
+	taskEdges, err := computeEdges(dag.Operators, operatorToTask)
+	if err != nil {
+		return nil, err
+	}
+
+	operatorMetadataPath := fmt.Sprintf("compile-airflow-metadata-%s", uuid.New().String())
+	operatorOutputPath := fmt.Sprintf("compile-airflow-output-%s", uuid.New().String())
+
+	defer func() {
+		go utils.CleanupStorageFiles(ctx, storageConfig, []string{operatorMetadataPath, operatorOutputPath})
+	}()
+
+	jobName := fmt.Sprintf("compile-airflow-operator-%s", uuid.New().String())
+	jobSpec := job.NewCompileAirflowSpec(
+		jobName,
+		storageConfig,
+		operatorMetadataPath,
+		operatorOutputPath,
+		dagId,
+		string(dag.Metadata.Schedule.CronSchedule),
+		taskToJobSpec,
+		taskEdges,
+	)
+
+	if err := jobManager.Launch(ctx, jobSpec.JobName(), jobSpec); err != nil {
+		return nil, err
+	}
+
+	jobStatus, err := job.PollJob(ctx, jobSpec.JobName(), jobManager, pollCompileAirflowInterval, pollCompileAirflowTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	if jobStatus == shared.FailedExecutionStatus {
+		return nil, errors.New("Compile Airflow job failed.")
+	}
+
+	var execState shared.ExecutionState
+	if err := utils.ReadFromStorage(
+		ctx,
+		storageConfig,
+		operatorMetadataPath,
+		&execState,
+	); err != nil {
+		return nil, err
+	}
+
+	if execState.Status == shared.FailedExecutionStatus {
+		return nil, errors.Newf("Compile Airflow job failed: %v \n logs: %v \n failure type: %v", execState.Error, execState.UserLogs, execState.FailureType)
+	}
+
+	airflowDagFile, err := storage.NewStorage(storageConfig).Get(ctx, operatorOutputPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the AirflowRuntimeConfig for `dag`
+	newRuntimeConfig := dag.EngineConfig
```

Comment:
here should we use the 'struct constructor' like
```
newRuntimeConfig := dag.EngineConfig{
  AirflowConfig: {
    OperatorToTask: ...
  }
}
```

Reply:
The `dag.EngineConfig` has already been initialized as part of the dag request, so we only need to write some of fields of `AirflowConfig`.